### PR TITLE
Test/middleware peers

### PR DESCRIPTION
**Pull request by zhouyuhe**
Test peer middleware

### CODE REVIEW

File: src/features/peers/redux/__test__/middleware.test.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      dispatch,
+      getState
+    };
+    next = jest.fn();
+    action = {
+      type: BOOTSTRAP
+    };
+  });
+
+  test("with correct payload", () => {
```
Comment (steinalex) [inline src/features/peers/redux/__test__/middleware.test.ts]: nit: Not sure this is a great test description...

File: src/features/peers/redux/__test__/middleware.test.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    const updateAction = {
+      payload: [{ name: "APPLE", symbol: "AAPL" }],
+      type: "UPDATE_TOP_PEERS"
+    };
+    topPeersMiddleware(mockSocket)(store)(next)(action);
+    expect(next).toHaveBeenCalled();
+    expect(on).toHaveBeenCalledWith(topPeers, expect.anything());
+    expect(getState).toHaveBeenCalledTimes(0);
+    expect(emit).toHaveBeenCalledTimes(0);
+    expect(dispatch).toHaveBeenCalledWith(updateAction);
```
Comment (steinalex) [inline src/features/peers/redux/__test__/middleware.test.ts]: q: Could this be split into a few tests?

Comment (zhouyuhe) [reply src/features/peers/redux/__test__/middleware.test.ts]: I've now changed it to small tests

File: src/features/peers/redux/__test__/middleware.test.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    store = {
+      getState: jest.fn(() => peerState),
+      dispatch: jest.fn()
+    };
+
+    next = jest.fn();
+  });
+
+  test.only("socket.on and dispatch been called", () => {
```
Comment (steinalex) [inline src/features/peers/redux/__test__/middleware.test.ts]: nit: Might be nicer to split your `socket` creation and `dispatch` tests into two tests?

File: src/features/peers/redux/middleware.ts [hunk 4]
```diff
@@ -4,13 +4,16 @@ import { SocketService } from "services/socketService";
 import { Middleware } from "redux";
 import { AppState } from "store";
 
-type Dependencies = {
+export type PeersDependencies = {
```
Comment (steinalex) [inline src/features/peers/redux/middleware.ts]: q: I see you're now exporting `PeersDependencies`, any reason for that?

Comment (zhouyuhe) [reply src/features/peers/redux/middleware.ts]: Previously I need it in my test, now I don't need to export it.
Will remove it

File: src/features/peers/redux/__test__/middleware.test.ts (hunk 3 above)
Comment (zhouyuhe) [reply src/features/peers/redux/__test__/middleware.test.ts]: I think my dispatch is within the socket, so I did it in one test

**Review by steinalex — APPROVED** [review]

**Pull request merged**